stl/inc/iterator

```diff
@@ -120,7 +120,13 @@ _NODISCARD insert_iterator<_Container> inserter(_Container& _Cont, typename _Con
 
 
 // CLASS TEMPLATE istream_iterator
+#if _HAS_CXX20
+template <class _Ty, class _Elem = char, class _Traits = char_traits<_Elem>, class _Diff = ptrdiff_t,
+    enable_if_t<conjunction_v<is_default_constructible<_Ty>, is_copy_constructible<_Ty>, is_copy_assignable<_Ty>>,

```

Comment:
I think we need to `static_assert` these properties with nice error messages, and we should do so in all language modes. These are not new requirements in C++20, they were always implicit. We definitely don't want to have the template parameters vary between C++20 and sub-20 modes.

Reply:
There's also no reason to enable_if_t this because it isn't like there's a primary template that would get used instead.